src/ComposerRequireChecker/ASTLocator/LocateASTFromFiles.php

```diff
@@ -39,11 +45,21 @@ public function __invoke(Traversable $files): Traversable
                 continue;
             }
 
-            try {
-                $stmts = $this->parser->parse(file_get_contents($file), $this->errorHandler);
-            } catch (Error $e) {
-                // Convert the parse error into one which has the file information
-                throw new FileParseFailed($file, $e);
+            $key = sha1_file($file);
```

Comment:
This hashing should occur one level up, when looking at directories: that would allow for easier cache slamming done implicitly by this tool.

The cache key should factor in the current `composer.lock` and `installed.json`, I'd say :thinking:

Reply:
The collected symbols for a given file are always the same, unless the file was modified, right?
So if we have the symbols in cache based on the hashed file contents, it can never be different. Or do I miss something.

Reply:
Depends on whether the code in this package (or any dependency that has to do with this package) changes behavior in any way.

Reply:
In the `SymbolCache` we can modify the `getKey` method and include other things.